### ISSUE TITLE
Script stopped working

### ISSUE THREAD

**Issue by dmaglinte77** [grahamr975/EWS-Office365-Contact-Sync#22]
This script stopped working for me at the end of september. MFA has been enabled in our tenant, but the user that I'm using to for the script has been excluded. But no matter what I do I cannot seem to get the script to work. Has anyone else seen this issue?
![Capture](https://user-images.githubusercontent.com/70035594/98042085-f124a800-1df0-11eb-9034-d6825513fce3.PNG)

**Comment by grahamr975** [comment]
Are you sure that the user is excluded? That portion of the script doesn't use application impersonation or EWS. I'd suggest you double-check the credentials. You can test by manually running the below code in a PowerShell window.

```
$Session = New-PSSession -ConfigurationName Microsoft.Exchange -ConnectionUri $ConnectionUri -Credential Get-Credential -Authentication Basic -AllowRedirection
Import-PSSession $Session -DisableNameChecking -AllowClobber
	
		# Import Global Address List into Powershell from Office 365 Exchange as an array
		$ContactList = Get-User -ResultSize unlimited
```

**Comment by dmaglinte77** [comment]
It’s erroring out. Here’s the error message:

[cid:image002.jpg@01D6B38A.225F54A0]

**Comment by grahamr975** [comment]
@dmaglinte77 I don't see the error message?

**Comment by dmaglinte77** [comment]
![ScriptError](https://user-images.githubusercontent.com/70035594/98294582-2d394380-1f7e-11eb-98b3-33e51631d666.PNG)

**Comment by grahamr975** [comment]
Yep, definitely looks like an authentication issue with the account you're using. Could be one or more of the below issues:
1. Invalid username/password
2. Basic authentication is disabled in your environment
3. The account lacks Exchange Online admin access

I'd suggest you follow Microsoft's troubleshooting guide [here ](https://docs.microsoft.com/en-us/exchange/troubleshoot/administration/access-denied-connect-powershell)and verify that basic authentication is enabled for the account.

**Comment by dmaglinte77** [comment]
Good morning!
I was able to make some progress. I had to go into the MFA settings inside Org Settings and allow Exchange Online Powershell to use Basic Auth. But now I'm getting different errorS:

PS>TerminatingError(Connect-EXCExchange): "Exception calling "AutodiscoverUrl" with "2" argument(s): "The Autodiscover service couldn't be located.""
>> TerminatingError(Connect-EXCExchange): "Exception calling "AutodiscoverUrl" with "2" argument(s): "The Autodiscover service couldn't be located.""
2020/11/06 10:57:08 ERROR Failed to Sync-ContactList for CTeska@ChoicesCCS.org Exception calling "AutodiscoverUrl" with "2" argument(s): "The Autodiscover service couldn't be located."
![AutodiscoverURLError](https://user-images.githubusercontent.com/70035594/98387212-20b8f780-201f-11eb-819d-d7a59f87b742.PNG)

**Comment by grahamr975** [comment]
Looks like you're using basic EWS authentication (Negotiate Auth), which your environment is probably also blocking. Can you try using the latest version of the script along with the **-ModernAuth** flag? Make sure to follow the new instructions found in the read me.

**Comment by dmaglinte77** [comment]
No dice. I re-downloaded your code and ran it again. Made sure -ModernAuth was set, but still getting that same error message.

**Comment by grahamr975** [comment]
@dmaglinte77 The newest version of the script now uses an AzureApp so it completely bypasses MFA-related issues. Please update let me now if this helps.

**Comment by dmaglinte77** [comment]
Will the old script still work or do we need to immediately start using this new version?
